datumaro/util/os_util.py

```diff
@@ -49,6 +66,57 @@ def walk(path, max_depth=None):
 
         yield dirpath, dirnames, filenames
 
+def copytree(src, dst):
+    # Shutil works very slow pre 3.8
+    # https://docs.python.org/3/library/shutil.html#platform-dependent-efficient-copy-operations
+    # https://bugs.python.org/issue33671
+
+    if sys.version_info[1] >= 8:
+        shutil.copytree(src, dst)
+        return
+
+    basedir = osp.dirname(dst)
+    if basedir:
+        os.makedirs(basedir, exist_ok=True)
+
+    if sys.platform == 'windows':
+        # Ignore
+        #   B603: subprocess_without_shell_equals_true
+        #   B607: start_process_with_partial_path
+        # In this case we control what is called and command arguments
+        # PATH overriding is considered low risk
+        subprocess.check_call(["xcopy", src, dst, "/s", "/e"]) # nosec
+    elif sys.platform == 'linux':
+        # As above
+        subprocess.check_call(["cp", "-r", "--reflink=auto", src, dst]) # nosec
+    else:
+        shutil.copytree(src, dst)
+
+@contextmanager
+def suppress_output(stdout: bool = True, stderr: bool = False):
+    with open(os.devnull, 'w') as devnull:
+        es = ExitStack()
+
+        if stdout:
+            es.enter_context(redirect_stdout(devnull))
+
+        if stderr:
+            es.enter_context(redirect_stderr(devnull))
```

Comment:
This part should be within the `with` block, otherwise any exceptions produced within `enter_context` or `redirect_stderr` may cause breakage.

Reply:
Fixed